ribs/emitters/_gradient_emitter.py

```diff
@@ -0,0 +1,297 @@
+            return self._initial_solutions
+
+        if self._jacobian_batch is None:
+            raise RuntimeError("Please call ask_dqd() and tell_dqd() "
+                               "before calling ask().")
+
+        if self._measure_gradients:
+            noise = self._rng.normal(
+                loc=0.0,
+                scale=self._sigma_g,
+                size=self._jacobian_batch.shape[:2],
+            )
+            noise[:, 0] = np.abs(noise[:, 0])
+            noise = np.expand_dims(noise, axis=2)
+            offsets = np.sum(np.multiply(self._jacobian_batch, noise), axis=1)
+            sols = offsets + self._parents
+        else:
+            # Transform the Jacobian
+            if len(self._jacobian_batch.shape) == 3:
+                self._jacobian_batch = np.squeeze(self._jacobian_batch[:,
+                                                                       0:1, :],
+                                                  axis=1)
+            sols = self._parents + self._jacobian_batch * self._sigma_g
+
+        return sols
+
+    def tell_dqd(self,
+                 solution_batch,
+                 objective_batch,
+                 measures_batch,
+                 jacobian_batch,
+                 status_batch,
+                 value_batch,
+                 metadata_batch=None):
+        """Gives the emitter results from evaluating the solutions.
+
+        Args:
+            solution_batch (array-like): (batch_size, :attr:`solution_dim`)
+                array of solutions generated by this emitter's
+                :meth:`ask_dqd()` method.
+            objective_batch (array-like): 1d array containing the objective
+                function value of each solution.
+            measures_batch (array-like): (batch_size, measure space dimension)
+                array with the measure space coordinates of each solution.
+            jacobian_batch (array-like): (batch_size, 1 + measure_dim,
+                solution_dim) array consisting of Jacobian matrices of the
+                solutions obtained from :meth:`ask_dqd`. Each matrix should
+                consist of the objective gradient of the solution followed by
+                the measure gradients.
+            status_batch (array-like): 1d array of
+                :class:`ribs.archive.addstatus` returned by a series of calls
+                to archive's :meth:`add()` method.
+            value_batch (array-like): 1d array of floats returned by a series
+                of calls to archive's :meth:`add()` method. for what these
+                floats represent, refer to :meth:`ribs.archives.add()`.
+            metadata_batch (array-like): 1d object array containing a metadata
+                object for each solution.
+        """
+        # preprocess + validate args
+        solution_batch = np.asarray(solution_batch)
+        objective_batch = np.asarray(objective_batch)
+        measures_batch = np.asarray(measures_batch)
```

Comment:
I think we also need to handle jacobian_batch here with asarray. Seems we forgot this in gradientarborscenceemitter.tell_dqd; could you add it there too?

Reply:
added validation on jacobian for  GradientOperatorEmitter (previously called GradientEmitter). So you want me do add the same check to GradientArboresenceEmitter as well?